apps/system/linux/machine/generic/platform_info.c

```diff
@@ -243,13 +243,13 @@ linux_proc_init(struct remoteproc *rproc,
 		return NULL;
 	}
 	prproc->shm_old_io = io;
-	shm = &prproc->shm;
-	shm->pa = 0;
-	shm->da = 0;
-	shm->size = prproc->shm_size;
-	metal_io_init(&prproc->shm_new_io, io->virt, &shm->pa,
-		      shm->size, -1, 0, &linux_proc_io_ops);
-	shm->io = &prproc->shm_new_io;
+
+	pa = 0;
+	metal_io_init(&prproc->shm_new_io, io->virt, &pa,
```

Comment:
&pa point to a non static variable, this seems not correct

Reply:
metal_io_init() is a bit odd here, all the users I see have to do something hacky to get a permanent memory pointer. It might be good to think about copying the data out of physmap to something internal to the metal_io_region struct. That way the local physmap can be freed after being passed into metal_io_init(). For now I'll rework this using the io ops.